docs/source/how-to-upstream.mdx

```diff
@@ -1,28 +1,117 @@
+
+```py
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.upload_file(path_or_fileobj="/path/to/local/folder/README.md", 
+...                 path_in_repo="README.md", 
+...                 repo_id="username/test-dataset",
+...                 repo_type="dataset",
+... )
+```
+
+### Upload a folder
+
+Use the [`upload_folder`] function to upload a local folder to an existing repository. Specify the path of the local folder to upload, where you want to upload the folder to in the repository, and the name of the repository you want to add the folder to. Depending on your repository type, you can optionally set the repository type as a `dataset`, `model` or `space`.
+
+```py
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.upload_folder(folder_path="/path/to/local/folder",
+...                   path_in_repo="my-dataset/train",
+...                   repo_id="username/test-dataset",
+...                   repo_type="dataset",
+... )
+# https://huggingface.co/datasets/username/test-dataset/tree/main/my-dataset/train/
+```
+
+### create_commit
+
+If you want to work at a commit-level, use the [`create_commit`] function directly. There are two types of operations supported by [`create_commit`]:
+
+- `CommitOperationAdd` uploads a file to the Hub. If the file already exists, the file contents are overwritten. This operation accepts two arguments:
+
+   * `path_in_repo`: the repository path to upload a file to.
+   * `path_or_fileobj`: either a path to a file on your filesystem or a file-like object. This is the content of the file to upload to the Hub.
+
+- `CommitOperationDelete` removes a file from a repository. This operation accepts `path_in_repo` as an argument.
+
+For example, if you want to upload two files and delete a file in a Hub repository:
+
+1. Use the appropriate `CommitOperation` to add and delete a file:
+
+```py
+>>> from huggingface_hub import HfApi, CommitOperationAdd, CommitOperationDelete
+>>> api = HfApi()
+>>> operations = [
+...     CommitOperationAdd(path_in_repo="LICENSE.md", path_or_fileobj="~/repo/LICENSE.md"),
+...     CommitOperationAdd(path_in_repo="weights.h5", path_or_fileobj="~/repo/weights-final.h5"),
+...     CommitOperationDelete(path_in_repo="old-weights.h5"),
+... ]
+```
+
+2. Pass your operations to [`create_commit`]:
+
+```py
+>>> api.create_commit(
+...     repo_id="lysandre/test-model",
+...     operations=operations,
+...     commit_message="Upload my model weights and license",
+... )
+```
+
+In addition to [`upload_file`] and [`upload_folder`], the following functions also use [`create_commit`] under the hood: 
+
+* [`delete_file`] deletes a single file from a repository on the Hub. 
+* [`metadata_update`] updates a repository's metadata.
+
+For more detailed information, take a look at the [`HfApi`] reference.
+
+## Push files with Git LFS
+
+For large files (>5GB), you need to install a custom transfer agent for Git LFS:
```

Comment:
All files with 10mb or more also use Git LFS, just that the `lfs-enable-largefiles` is needed for very very large files. Maybe mention that all methods will automatically add your files to git lfs if needed, and just use lfs-enable-largefiles for these very large files